scripts/irods/test/test_dynamic_peps.py

```diff
@@ -492,3 +492,59 @@ def test_if_Genquery2Input_is_exposed__issue_7570(self):
                 self.user.run_icommand(['imeta', 'rm', '-C', self.user.session_collection, f'{avu_prefix}_1', self.user.zone_name])
                 self.user.run_icommand(['imeta', 'rm', '-C', self.user.session_collection, f'{avu_prefix}_2', '0'])
                 self.user.run_icommand(['imeta', 'rm', '-C', self.user.session_collection, f'{avu_prefix}_3', '0'])
+
+    @unittest.skipIf(plugin_name != 'irods_rule_engine_plugin-irods_rule_language' or test.settings.RUN_IN_TOPOLOGY, "Requires NREP and single node zone.")
+    def test_if_StructFileExtAndRegInp_is_exposed__issue_7413(self):
+        with temporary_core_file() as core:
+            avu_prefix = 'test_if_StructFileExtAndRegInp_is_exposed__issue_7413'
+
+            # Add a rule to core.re which when triggered will add AVUs to the user's
+            # session collection.
+            core.add_rule(dedent('''
+            pep_api_struct_file_ext_and_reg_post(*INSTANCE, *COMM, *STRUCTINP) {{
+                writeLine("serverLog",*STRUCTINP);
+                *v = *STRUCTINP.objPath;
+                msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{avu_prefix}_0', *v, '');
+
+                *v = *STRUCTINP.collection;
+                msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{avu_prefix}_1', *v, '');
+
+                *w = *STRUCTINP.oprType;
+                msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{avu_prefix}_2', *w, '');
+            }}
+            '''.format(**locals())))
+
+            TEST_COLLECTION_NAME = "%s/%s" % (self.user.session_collection, 'testcoll')
+            TEST_TARFILE_NAME = "%s/%s" % (self.user.session_collection, 'test.tar')
+
+            try:
+                self.user.assert_icommand(['imkdir', TEST_COLLECTION_NAME], 'STDOUT')
+                # Make some test files...
+                for x in range(0, 5):
+                    self.user.assert_icommand(['itouch', "%s/file%s" % (TEST_COLLECTION_NAME, x)], 'STDOUT')
+
+                self.user.assert_icommand(['ibun', '-cDtar', "%s/%s" % (self.user.session_collection, 'test.tar'), TEST_COLLECTION_NAME], 'STDOUT')
+
+                self.user.assert_icommand(['imkdir', TEST_COLLECTION_NAME + "_backup"], 'STDOUT')
```

Comment:
Same here. If the icommand doesn't produce any output on success, `STDOUT` isn't needed.